src/apps/weblib/ts-api/plugins/shorthands.ts

```diff
@@ -201,26 +201,78 @@ export class Shorthands implements Plugin {
 			channel = [channel]
 		}
 
-		if (channel === null || Array.isArray(channel)) {
-			channel = { set: channel }
+		if (this._isAnySeriesList(channel)) {
+			channel = { set: this._normalizeSeriesList(channel) }
+		} else if (channel.set !== undefined && this._isAnySeriesList(channel.set)) {
+			channel.set = this._normalizeSeriesList(channel.set)
 		}
 
-		if (typeof channel?.attach === 'string') {
-			channel.attach = [channel.attach]
+		if (channel.attach !== undefined && this._isAnySeriesList(channel.attach)) {
+			const attach = this._normalizeSeriesList(channel.attach)
+			if (attach !== null) {
+				channel.attach = attach
+			} else {
+				delete channel.attach
+			}
 		}
 
-		if (typeof channel?.detach === 'string') {
-			channel.detach = [channel.detach]
+		if (channel.detach !== undefined && this._isAnySeriesList(channel.detach)) {
+			const detach = this._normalizeSeriesList(channel.detach)
+			if (detach !== null) {
+				channel.detach = detach
+			} else {
+				delete channel.detach
+			}
 		}
 
-		if (typeof channel?.set === 'string') {
-			channel.set = [channel.set]
+		return channel
+	}
+
+	private _isAnySeriesList(value: unknown): value is AnySeriesList {
+		return (
+			value === null ||
+			typeof value === 'string' ||
+			Array.isArray(value) ||
+			(typeof value === 'object' && 'name' in value)
+		)
+	}
+
+	private _normalizeSeriesList(seriesList: AnySeriesList): Data.SeriesList | null {
+		if (seriesList === null) {
+			return null
 		}
+		if (Array.isArray(seriesList)) {
+			if (seriesList.length === 0) return null
+			else
+				return seriesList.map((seriesDescriptor) =>
+					this._normalizeSeriesDescriptor(seriesDescriptor)
+				)
```

Comment:
Only semantic: if the condition is formatted over several lines, it is not usual to leave out the brackets.

Reply:
Köszi, ezt most igy hagyom, mert a libben a kód formázást a prettierrel csináljuk, ott kellene majd beállítani rule-t erre.